tests/pipeline/test_statistical.py

```diff
@@ -0,0 +1,759 @@
+            )
+        with self.assertRaises(TypeError):
+            returns.spearmanr(
+                target=date_factor_slice,
+                correlation_length=correlation_length,
+            )
+
+    @parameter_space(returns_length=[2, 3], regression_length=[3, 4])
+    def test_factor_regression_method(self, returns_length, regression_length):
+        """
+        Ensure that `Factor.linear_regression` is consistent with the built-in
+        factor `RollingLinearRegressionOfReturns`.
+        """
+        my_asset = self.my_asset
+        start_date = self.pipeline_start_date
+        end_date = self.pipeline_end_date
+        run_pipeline = self.run_pipeline
+
+        returns = Returns(window_length=returns_length, inputs=[self.col])
+        returns_slice = returns[my_asset]
+
+        regression = returns.linear_regression(
+            target=returns_slice, regression_length=regression_length,
+        )
+        expected_regression = RollingLinearRegressionOfReturns(
+            target=my_asset,
+            returns_length=returns_length,
+            regression_length=regression_length,
+        )
+
+        # This built-in constructs its own Returns factor to use as an input,
+        # so the only way to set our own input is to do so after the fact. This
+        # should not be done in practice. It is necessary here because we want
+        # Returns to use our random data as an input, but by default it is
+        # using USEquityPricing.close.
+        expected_regression.inputs = [returns, returns_slice]
+
+        columns = {
+            'regression': regression,
+            'expected_regression': expected_regression,
+        }
+
+        results = run_pipeline(Pipeline(columns=columns), start_date, end_date)
+        regression_results = results['regression'].unstack()
+        expected_regression_results = results['expected_regression'].unstack()
+
+        assert_frame_equal(regression_results, expected_regression_results)
+
+        # Make sure we cannot call the linear regression method on factors or
+        # slices of dtype `datetime64[ns]`.
+        class DateFactor(CustomFactor):
+            window_length = 1
+            inputs = []
+            dtype = datetime64ns_dtype
+            window_safe = True
+
+            def compute(self, today, assets, out):
+                pass
+
+        date_factor = DateFactor()
+        date_factor_slice = date_factor[my_asset]
+
+        with self.assertRaises(TypeError):
+            date_factor.linear_regression(
+                target=returns_slice, regression_length=regression_length,
+            )
+        with self.assertRaises(TypeError):
+            returns.linear_regression(
+                target=date_factor_slice, regression_length=regression_length,
+            )
+
+    @parameter_space(correlation_length=[1, 2, 3, 4])
+    def test_factor_correlation_methods_two_factors(self, correlation_length):
```

Comment:
This test and `test_factor_regression_method_two_factors` below are the only new tests for the factor-to-factor computations. @ssanderson Previously I had tried to factor out some of the shared code here with `test_correlation_factors` above, but that proved to be too confusing for what it was worth, so this test contains similar logic to the pre-existing tests. It might be worth a glance, but there's not much new here.